sale_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,649 @@
+            ("draft", "Draft"),
+            ("open", "Open"),
+            ("done", "Done"),
+            ("expired", "Expired"),
+        ],
+        compute="_compute_state",
+        store=True,
+        copy=False,
+    )
+    validity_date = fields.Date(readonly=True, states={"draft": [("readonly", False)]})
+    client_order_ref = fields.Char(
+        string="Customer Reference",
+        copy=False,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    note = fields.Text(readonly=True, states={"draft": [("readonly", False)]})
+    user_id = fields.Many2one(
+        "res.users",
+        string="Salesperson",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    team_id = fields.Many2one(
+        "crm.team",
+        string="Sales Team",
+        change_default=True,
+        default=_get_default_team,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=_default_company,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    sale_count = fields.Integer(compute="_compute_sale_count")
+
+    fiscal_position_id = fields.Many2one(
+        "account.fiscal.position", string="Fiscal Position"
+    )
+
+    amount_untaxed = fields.Monetary(
+        string="Untaxed Amount",
+        store=True,
+        readonly=True,
+        compute="_compute_amount_all",
+        track_visibility="always",
+    )
+    amount_tax = fields.Monetary(
+        string="Taxes", store=True, readonly=True, compute="_compute_amount_all"
+    )
+    amount_total = fields.Monetary(
+        string="Total", store=True, readonly=True, compute="_compute_amount_all"
+    )
+
+    # Fields use to filter in tree view
+    original_uom_qty = fields.Float(
+        string="Original quantity",
+        compute="_compute_uom_qty",
+        search="_search_original_uom_qty",
+        default=0.0,
+    )
+    ordered_uom_qty = fields.Float(
+        string="Ordered quantity",
+        compute="_compute_uom_qty",
+        search="_search_ordered_uom_qty",
+        default=0.0,
+    )
+    invoiced_uom_qty = fields.Float(
+        string="Invoiced quantity",
+        compute="_compute_uom_qty",
+        search="_search_invoiced_uom_qty",
+        default=0.0,
+    )
+    remaining_uom_qty = fields.Float(
+        string="Remaining quantity",
+        compute="_compute_uom_qty",
+        search="_search_remaining_uom_qty",
+        default=0.0,
+    )
+    delivered_uom_qty = fields.Float(
+        string="Delivered quantity",
+        compute="_compute_uom_qty",
+        search="_search_delivered_uom_qty",
+        default=0.0,
+    )
+
+    def _get_sale_orders(self):
+        return self.mapped("line_ids.sale_lines.order_id")
+
+    @api.depends("line_ids")
+    def _compute_line_count(self):
+        self.line_count = len(self.mapped("line_ids"))
```

Comment:
```suggestion
        self.line_count = len(self.line_ids)
```
seems you don't need to mapp